presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java

```diff
@@ -535,9 +540,13 @@ public static Slice realToLongDecimal(long value, long precision, long scale, Bi
     private static Slice realToLongDecimal(long value, long precision, long scale)
     {
         float floatValue = intBitsToFloat(intScale(value));
+        if (Float.isInfinite(floatValue) || Float.isNaN(floatValue)) {
```

Comment:
add TODO for faster implementation

Reply:
I don't think this be revisited unless benchmarks show we should do it. We could place such TODOs in quite a few places in the code, would that be useful?

Reply:
I added them - they are no harm. Maybe it will motivate sb to do the benchmarks in the future.